### PR TITLE
s3 sample

### PR DESCRIPTION
**Pull request by graebm**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samples/s3/pom.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.11</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
```
Comment (DavidOgunsAWS) [inline samples/s3/pom.xml:31]: I don't think anything under this tag is non-default. This plugins are assumed by maven and will be included and associated with the tool itself. Try removing it and everything should work just fine.

File: samples/s3/src/main/java/com/example/s3/PutGet.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            }
+            System.out.println("Upload complete");
+
+            System.out.println("Downloading...");
+            GetObjectRequest getRequest = GetObjectRequest.builder().bucket(bucket).key(key).build();
+            GetDataConsumer getDataConsumer = new PutGet.GetDataConsumer(getFilepath);
+            CompletableFuture<GetObjectOutput> getFuture = s3Client.getObject(getRequest, getDataConsumer);
+            try {
+                getFuture.get();
+            } catch (InterruptedException | ExecutionException e) {
```
Comment (DavidOgunsAWS) [inline samples/s3/src/main/java/com/example/s3/PutGet.java:76]: extreme nit: ExecutionExceptions usually wrap the relevant exception in the causedBy and don't reveal anything itself. The full stack trace will still show what matters, but usually you can unwrap those once before rewrapping in RuntimeException and throwing that.

**Review by DavidOgunsAWS — APPROVED** [review]

File: samples/s3/pom.xml (hunk 1 above)
Comment (graebm) [reply samples/s3/pom.xml:31]: Cool. I cut out EVERY plugin, and it still seems to work.

These all came from letting maven create an "archetype" project for me.

File: samples/s3/src/main/java/com/example/s3/PutGet.java (hunk 2 above)
Comment (graebm) [reply samples/s3/src/main/java/com/example/s3/PutGet.java:76]: I'm going to keep it as is, for the sake of keeping the sample as simple as possible. If the stacktrace shows the issue, that's sufficient

Thanks for the info, though

**Review by rccarper — APPROVED** [review]

**Pull request merged**